source/header.c

```diff
@@ -342,12 +345,18 @@ int aws_cryptosdk_priv_hdr_parse_alg_suite_data(
 }
 
 int aws_cryptosdk_priv_hdr_parse_iv(struct aws_cryptosdk_hdr *hdr, uint8_t iv_len, struct aws_byte_cursor *cur) {
+    AWS_PRECONDITION(aws_cryptosdk_hdr_is_valid(hdr));
+    AWS_PRECONDITION(&hdr->alloc != NULL);
```

Comment:
You don't need `&hdr->alloc`, right? Why not just `hdr->alloc != NULL`?

Reply:
See above!